File: app/src/main/java/com/kickstarter/libs/preferences/StringPreferenceType.java
package com.kickstarter.libs.preferences;

public interface StringPreferenceType {
  /**
   * Get the current value of the preference.
   */
  String get();

  /**
   * Returns whether a value has been explicitly set for the preference.
   */
  boolean isSet();

  /**
   * Set the preference to a value.
   */
  void set(String value);

  /**
   * Delete the currently stored preference.
   */
  void delete();
}

File: app/src/main/java/com/kickstarter/libs/preferences/StringPreferenceType.kt
package com.kickstarter.libs.preferences

interface StringPreferenceType {
    /**
     * Get the current value of the preference.
     */
    fun get(): String?

    /**
     * Returns whether a value has been explicitly set for the preference.
     */
    val isSet: Boolean

    /**
     * Set the preference to a value.
     */
    fun set(value: String?)

    /**
     * Delete the currently stored preference.
     */
    fun delete()
}
